syntax = "proto3";
package logtail.models;

import "log_event.proto";
import "metric_event.proto";
import "span_event.proto";

message PipelineEventGroup {  
    map<string, bytes> Metadata = 1;  
    map<string, bytes> Tags = 2;  
    message LogEvents {
        repeated LogEvent Events = 1;
    }
    message MetricEvents {
        repeated MetricEvent Events = 1;
    }
    message SpanEvents {
        repeated SpanEvent Events = 1;
    }
    oneof PipelineEvents {
        LogEvents Logs = 3;
        MetricEvents Metrics = 4;
        SpanEvents Spans = 5;
    }
}
